src/qt/managenamespage.cpp

```diff
@@ -0,0 +1,143 @@
+#include <qt/forms/ui_managenamespage.h>
+
+#include <qt/csvmodelwriter.h>
+#include <qt/guiutil.h>
+#include <qt/nametablemodel.h>
+#include <qt/platformstyle.h>
+#include <qt/walletmodel.h>
+
+#include <QMessageBox>
+#include <QMenu>
+#include <QSortFilterProxyModel>
+
+ManageNamesPage::ManageNamesPage(const PlatformStyle *platformStyle, QWidget *parent) :
+    QWidget(parent),
+    platformStyle(platformStyle),
+    ui(new Ui::ManageNamesPage),
+    model(nullptr),
+    walletModel(nullptr),
+    proxyModel(nullptr)
+{
+    ui->setupUi(this);
+
+    // Context menu actions
+    QAction *copyNameAction = new QAction(tr("Copy &Name"), this);
+    QAction *copyValueAction = new QAction(tr("Copy &Value"), this);
+
+    // Build context menu
+    contextMenu = new QMenu();
+    contextMenu->addAction(copyNameAction);
+    contextMenu->addAction(copyValueAction);
+
+    // Connect signals for context menu actions
+    connect(copyNameAction, &QAction::triggered, this, &ManageNamesPage::onCopyNameAction);
+    connect(copyValueAction, &QAction::triggered, this, &ManageNamesPage::onCopyValueAction);
+
+    connect(ui->tableView, &QTableView::customContextMenuRequested, this, &ManageNamesPage::contextualMenu);
+    ui->tableView->setEditTriggers(QAbstractItemView::NoEditTriggers);
+
+    ui->tableView->installEventFilter(this);
+}
+
+ManageNamesPage::~ManageNamesPage()
+{
+    delete ui;
+}
+
+void ManageNamesPage::setModel(WalletModel *walletModel)
+{
+    this->walletModel = walletModel;
+    model = walletModel->getNameTableModel();
+
+    proxyModel = new QSortFilterProxyModel(this);
+    proxyModel->setSourceModel(model);
+    proxyModel->setDynamicSortFilter(true);
+    proxyModel->setSortCaseSensitivity(Qt::CaseInsensitive);
+    proxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
+
+    ui->tableView->setModel(proxyModel);
+    ui->tableView->sortByColumn(0, Qt::AscendingOrder);
+
+    ui->tableView->horizontalHeader()->setHighlightSections(false);
+
+    // Set column widths
+    ui->tableView->horizontalHeader()->resizeSection(
+            NameTableModel::Name, 320);
+    ui->tableView->horizontalHeader()->setSectionResizeMode(QHeaderView::Stretch);
+
+    connect(ui->tableView->selectionModel(), &QItemSelectionModel::selectionChanged,
+            this, &ManageNamesPage::selectionChanged);
+
+    selectionChanged();
+}
+
+bool ManageNamesPage::eventFilter(QObject *object, QEvent *event)
+{
+    if (event->type() == QEvent::FocusIn)
+    {
+        if (object == ui->tableView)
+        {
+        }
+    }
+    return QWidget::eventFilter(object, event);
+}
+
+void ManageNamesPage::selectionChanged()
+{
+    // Set button states based on selected tab and selection
+    QTableView *table = ui->tableView;
+    if (!table->selectionModel())
+        return;
+
+    const bool state = table->selectionModel()->hasSelection();
+    //ui->configureNameButton->setEnabled(state);
+    //ui->renewNameButton->setEnabled(state);
```

Comment:
I think you should remove the commented out lines here, and just add them in with the future changes that allow renewing names.  This is not directly related to this PR.  (But if you really prefer to keep them in, that's also fine with me.)

Reply:
Moved this code to https://github.com/namecoin/namecoin-core/pull/376